src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1083,6 +1083,15 @@
   <data name="DoNotCreateTasksWithoutPassingATaskSchedulerMessage" xml:space="preserve">
     <value>Do not create tasks without passing a TaskScheduler</value>
   </data>
+  <data name="DoNotDefineFinalizersForTypesDerivedFromMemoryManagerDescription" xml:space="preserve">
+    <value>Adding a finalizer for a type derived from MemoryManager&lt;T&gt; creates the possibility that the memory will be freed while it is still in use by a Span&lt;T&gt;.</value>
+  </data>
+  <data name="DoNotDefineFinalizersForTypesDerivedFromMemoryManagerMessage" xml:space="preserve">
+    <value>Adding a finalizer for a type derived from MemoryManager&lt;T&gt; creates the possibility that the memory will be freed while it is still in use by a Span&lt;T&gt;.</value>
+  </data>
+  <data name="DoNotDefineFinalizersForTypesDerivedFromMemoryManagerTitle" xml:space="preserve">
+    <value>Do not define finalizers for types derived from MemoryManager&lt;T&gt;</value>
+  </data>
```

Comment:
```suggestion
  <data name="DoNotDefineFinalizersForTypesDerivedFromMemoryManagerDescription" xml:space="preserve">
    <value>Adding a finalizer to a type derived from MemoryManager&lt;T&gt; may permit memory to be freed while it is still in use by a Span&lt;T&gt;.</value>
  </data>
  <data name="DoNotDefineFinalizersForTypesDerivedFromMemoryManagerMessage" xml:space="preserve">
    <value>Adding a finalizer to a type derived from MemoryManager&lt;T&gt; may permit memory to be freed while it is still in use by a Span&lt;T&gt;.</value>
  </data>
  <data name="DoNotDefineFinalizersForTypesDerivedFromMemoryManagerTitle" xml:space="preserve">
    <value>Do not define finalizers for types derived from MemoryManager&lt;T&gt;</value>
  </data>
```

Reply:
I have changed from `Adding a finalizer **to** a type derived from` into `Adding a finalizer **for** a type derived from` by @danmosemsft's comment, and it sounded better for me. Not sure if we should update it back, maybe follow the issue wording written by @terrajobst

Reply:
Adding x "for" y suggests you can do it externally to y, whereas adding x "to" y suggests it's part of y, which is why I think "to" is better wording here... a finalizer is part of the type, rather than something that can be done externally.

Reply:
I agree "to" is better - if I suggested "for", I must have been confused. I can't see my comment for some reason.

Reply:
What about "**on** a type"?

@danmosemsft - the comment was on a draft PR in @buyaa-n's fork.  No worries though. 😄

Reply:
That works, too.

Reply:
@danmosemsft comment was for title `Do not define finalizers for types derived from MemoryManager&lt;T&gt;`, my bad that I had changed it for all similar 😛